### PR TITLE
New features: Hexadecimal integers and array initializers

### PR DESCRIPTION
**Pull request by DavidCizek**
Hexadecimal integers and array initializers

### CODE REVIEW

File: test/System.Linq.Dynamic.Core.Tests/ExpressionTests.cs [hunk 1]
```diff
@@ -638,6 +638,29 @@ public void ExpressionTests_IntegerQualifiers_Exceptions()
         }
 
         [Fact]
+        public void ExpressionTests_HexadecimalInteger()
```
Comment (StefH) [inline test/System.Linq.Dynamic.Core.Tests/ExpressionTests.cs]: Move this test method to the correct place in the file. (All methods are sorted alphabetically.)

File: test/System.Linq.Dynamic.Core.Tests/ExpressionTests.cs [hunk 2]
```diff
@@ -1028,5 +1051,22 @@ public void ExpressionTests_MethodCall_ReferenceTypeToObjectParameter()
             //Assert
             Assert.Equal(expectedResult.Count(), result.Count());
         }
+
+        [Fact]
+        public void ExpressionTests_ArrayInitializer()
```
Comment (StefH) [inline test/System.Linq.Dynamic.Core.Tests/ExpressionTests.cs]: Move this test method to the correct place in the file. (All methods are sorted alphabetically.)

File: src/System.Linq.Dynamic.Core/Tokenizer/TextParser.cs [hunk 3]
```diff
@@ -380,5 +396,21 @@ private static TokenId GetAliasedTokenId(TokenId t, string alias)
...
+        {
+            if (char.IsDigit(c))
+            {
+                return true;
+            }
+
+            if (c <= '\x007f')
+            {
+                c |= (char)0x20;
+                return c >= 'a' && c <= 'f';
```
Comment (StefH) [inline src/System.Linq.Dynamic.Core/Tokenizer/TextParser.cs]: Will this also work for 'A' and 'F' (capital ?)

File: src/System.Linq.Dynamic.Core/Tokenizer/TextParser.cs [hunk 4]
```diff
@@ -290,6 +290,17 @@ public void NextToken()
                             NextChar();
                         } while (char.IsDigit(_ch));
 
+                        bool hexInteger = false;
```
Comment (StefH) [inline src/System.Linq.Dynamic.Core/Tokenizer/TextParser.cs]: Code coverage from `TextParser.cs` dropped a bit, can you please check if you can add some more specific tests in this area?

File: test/System.Linq.Dynamic.Core.Tests/ExpressionTests.cs [hunk 5]
```diff
@@ -1028,5 +1051,22 @@ public void ExpressionTests_MethodCall_ReferenceTypeToObjectParameter()
...
+        {
+            //Arrange
+            var list = new[] { 0, 1, 2, 3, 4 };
+
+            //Act
+            var result1 = list.AsQueryable().SelectMany("new[] {}");
```
Comment (StefH) [inline test/System.Linq.Dynamic.Core.Tests/ExpressionTests.cs]: Maybe add a test to test array in array ?
Like `SelectMany("new[] { new[] { it + 1, it + 2 }, new[] { 10, 20 } }");`

File: src/System.Linq.Dynamic.Core/Tokenizer/TextParser.cs (hunk 3 above)
Comment (DavidCizek) [inline src/System.Linq.Dynamic.Core/Tokenizer/TextParser.cs]: Yes. Binary or with 0x20 makes all letters lowercase.

**Comment by DavidCizek** [comment]
Done

**Pull request merged**